### DISCUSSION TITLE
endless loop only on small intel cpus

### DISCUSSION THREAD

**Discussion by DirkHunniger** [ImageMagick/ImageMagick#7087] [General]
Hi,

I am trying to convert a svg image to a png image with ImageMagick.
The testcase goes like this:

wget https://upload.wikimedia.org/wikipedia/commons/5/5f/Chromium_-_world_production_trend.svg
convert -resize 1250x -background "#ffffff00" Chromium_-_world_production_trend.svg PNG:out.png

On some computers the convert command goes into an endless loop on others it doesn't. I tryed with my desktop and a server on wmflabs. Both work without any problem. But when I try it on one of my two notebooks or on my "mini pc", it goes into an endless loop. 

dirk@dirk-System-Product-Name:~$ convert -version
Version: ImageMagick 6.9.11-60 Q16 x86_64 2021-01-25 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC Modules OpenMP(4.5) 
Delegates (built-in): bzlib djvu fftw fontconfig freetype heic jbig jng jp2 jpeg lcms lqr ltdl lzma openexr pangocairo png tiff webp wmf x xml zlib

I think it could be an error in the cpu. Could anyone confirm the problem.

Yours Dirk

**Comment by DirkHunniger** [comment]
I could simplify the svg causing the issue. You have to remame it from testcase.svg.txt to testcase.svg

[testcase.svg.txt](https://github.com/ImageMagick/ImageMagick/files/14156703/testcase.svg.txt)

**Comment by snibgo** [comment]
I suspect the problem is in a specific SVG rasterizer, rather than the CPU.

What software is rasterizing your SVG?

With IM v7.1.1-20 on Windows 11, if I use `msvg:`, I also get an infinite loop.
```
magick -verbose -debug all msvg:testcase.svg x.png
```
The debug output shows a repetition of:
```
pop graphic-context
text 0,0 "Year"
pop graphic-context
text 0,0 "Year"
pop graphic-context
text 0,0 "Year"
```
Using a prefix of `rsvg:`, or no prefix and using Inkscape, there is no problem.

**Comment by urban-warrior** [comment]
Thank you for reporting the issue. We have successfully reproduced it and are actively working on a patch to resolve it. You can expect this patch to be merged into the [main GIT branch](https://github.com/ImageMagick/ImageMagick), later today. As part of our commitment to quality, this fix will also be included in the upcoming [beta releases of ImageMagick](https://imagemagick.org/archive/beta/) by tomorrow. Your patience and feedback are greatly appreciated.
